community/kernel/src/main/java/org/neo4j/kernel/recovery/LatestCheckPointFinder.java

```diff
@@ -91,9 +93,28 @@ public LatestCheckPoint find( long fromVersionBackwards ) throws IOException
 
             if ( latestCheckPoint != null )
             {
+                // Is the latest start entry in this log file version later than what the latest check point targets?
+                LogPosition target = latestCheckPoint.getLogPosition();
                 boolean commitsAfterCheckPoint = latestStartEntry != null &&
-                        latestStartEntry.getStartPosition().compareTo( latestCheckPoint.getLogPosition() ) >= 0;
-                return new LatestCheckPoint( latestCheckPoint, commitsAfterCheckPoint , oldestVersionFound );
+                        latestStartEntry.getStartPosition().compareTo( target ) >= 0;
+                if ( !commitsAfterCheckPoint )
+                {
+                    if ( target.getLogVersion() < version )
+                    {
+                        // This check point entry targets a previous log file.
+                        // Go there and see if there's a transaction. Reader is capped to that log version.
+                        commitsAfterCheckPoint = extractFirstTxIdAfterPosition( target, target.getLogVersion() ) !=
```

Comment:
The intention of the code is to start at the target (in your example tx1) and search from there up until max where it is now (including). Isn't that what you're seeing in the code and its tests?